plutus-tx/README.md

```diff
@@ -1,3 +1,228 @@
+
+This package just provides support for PlutusTx, if you are looking for support for
+writing smart contracts using PlutusTx, please look in `wallet-api`.
+
+## Haskell language support
+
+In general, most "straightforward" Haskell should work. More advanced features may
+or may not work, depending on whether they rely on features of GHC Core that aren't
+supported. Most syntactic language extensions should be fine, you may get into trouble
+with more advanced type system extensions.
+
+TODO: link to GitHub issues for planned items once we're using them.
+
+Not supported, but support planned:
+- Mutually recursive datatypes (support planned)
+    - Self-recursive datatypes are supported
+- Record selectors (support planned)
+- Functions defined outside the PlutusTx expression
+
+Not supported, and support not planned:
+- Datatypes beyond simple "Haskell98" datatypes
+    - GADTs, constrained constructors, etc.
+- Abstract datatypes
+- Numeric types other than `Int`
+- Literal patterns
+- Typeclasses
+    - Some `Num`, `Eq`, and `Ord` methods on
+      `Int` and `Bytestring` are supported specially.
+- Anything involving coercions
+
+## Tutorial
+
+Let's write some code!
+
+```haskell
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE ScopedTypeVariables       #-}
+import Language.PlutusTx.TH
+import Language.PlutusTx.Lift
+import Language.PlutusCore
+import Language.PlutusCore.Pretty
+import Language.PlutusCore.Quote
+import Language.PlutusCore.Evaluation.CkMachine
+import qualified Language.Haskell.TH as TH
+import Data.Functor
+```
+
+### Writing basic PlutusTx programs
+
+The `PlcCode` type is an opaque type which contains the serialized Plutus Core code
+corresponding to a Haskell expression. The `plutus` function takes a typed Template Haskell
+`Q (TExp a)`, for any a, and produces a `Q (TExp PlcCode)`, which we then
+have to splice into our program.
+
+The fact that `plutus` takes a TH quote means that what you write inside the quote
+is *just normal Haskell* - there is no PlutusTx-specific syntax, and the PlutusTx compiler will
+tell you if you use any Haskell features which are not supported.
+
+Here's the most basic program we can write: one that just evaluates to the integer `1`.
+
+```haskell
+-- |
+-- >>> prettyPlcDef $ getAst integerOne
+-- (program 1.0.0
+--   (con 8 ! 1)
+-- )
+integerOne :: PlcCode
+-- We don't like unbounded integers in Plutus Core, so we have to pin
+-- down that numeric literal to an `Int` not an `Integer`.
+integerOne = $$(plutus [|| (1 :: Int) ||])
+```
+
+The Plutus Core program will look incomprehensible, which is fine, since you
+mostly won't want to look at the output of the compiler. However, it's instructive to
+look at it here just to get a vague idea of what's going on.
+
+We can already see a few features of Plutus Core here:
+- We have a *versioned* program. This ensures that we always know how to handle
```

Comment:
Clarify that this is the version of the language, not of the program itself (that is, it has nothing to do with "different versions of a smart contract")